langkit/expressions/structs.py

```diff
@@ -227,21 +227,36 @@ def __init__(self, struct_type, assocs, result_var_name=None,
                      abstract_expr=None):
             """
             :type struct_type: CompiledType
-            :type assocs: {names.Name: ResolvedExpression}
+            :type assocs: {names.Name|langkit.compiled_typesAbstractNodeData:
+                           ResolvedExpression}
             :type result_var_name: str|None
             :param AbstractExpression|None abstract_expr: See
                 ResolvedExpression's constructor.
             """
             self.static_type = struct_type
-            self.assocs = assocs
+
+            # For convenience, automatically convert names to the corresponding
```

Comment:
Sorry, I thought that was obvious: if instead of calling StructExpr’s constructor with AbstractNodeData instances, we could instead just pass the field name, having StructExpr automatically fetch the corresponding AbstractNodeData, that would be more convenient for callers. I’ll amend the comment.

Reply:
Thanks! Got it now